WordPress/src/main/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthFragment.kt

```diff
@@ -25,41 +35,71 @@ import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Content
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Error
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Loading
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Scanning
+import org.wordpress.android.ui.qrcodeauth.compose.state.ContentState
+import org.wordpress.android.ui.qrcodeauth.compose.state.ErrorState
+import org.wordpress.android.ui.qrcodeauth.compose.state.LoadingState
 import org.wordpress.android.ui.utils.UiHelpers
 import org.wordpress.android.viewmodel.observeEvent
 import javax.inject.Inject
 
 @AndroidEntryPoint
 @Suppress("TooManyFunctions")
-class QRCodeAuthFragment : Fragment(R.layout.qrcodeauth_fragment) {
+class QRCodeAuthFragment : Fragment() {
     @Inject lateinit var uiHelpers: UiHelpers
 
-    private val viewModel: QRCodeAuthViewModel by viewModels()
+    private val qrCodeAuthViewModel: QRCodeAuthViewModel by viewModels()
     private val dialogViewModel: BasicDialogViewModel by activityViewModels()
 
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View = ComposeView(requireContext()).apply {
+        setContent {
+            AppTheme {
+                QRCodeAuthScreen()
+            }
+        }
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        with(QrcodeauthFragmentBinding.bind(view)) {
-            initBackPressHandler()
-            observeViewModel()
-            initViewModel(savedInstanceState)
+        initBackPressHandler()
+        initViewModel(savedInstanceState)
+        observeViewModel()
+    }
+
+    @Composable
+    private fun QRCodeAuthScreen(viewModel: QRCodeAuthViewModel = viewModel()) {
```

Comment:
I'd move this out of the fragment 👍

Reply:
I agree, and I'd say define a convention for where to define Composables, and hot to name the classes, it'll help keeping things consistent.

Reply:
Since this is exclusive to the `QRCodeAuth` fragment and based on my experience with different Compose codebases I tend to place such composables in the same file as the class using them, making the composable function private.

Idk if this resonates with your view, I didn't find such usages in `woocommerce-android` but saw it in a lot of repos (plus it's an approach I'm using in a compose-desktop app without issues so far).

My current suggestion to solve this is visible in this commit: [67627b0](https://github.com/wordpress-mobile/WordPress-Android/pull/16934/commits/67627b03fc55028d046070f284679fe8e92d7615).

Reply:
> plus it's an approach I'm using in a compose-desktop app without issues so far

In a `compose-desktop` app you won't have a fragment, or am I missing something?

in `woocommerce-android` we are following an approach similar to the official compose [samples](https://github.com/android/compose-samples/tree/e6994123804b976083fa937d3f5bf926da4facc5/Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/signinsignup), we define the fragment in a file, then the Screen Composable in a separate file, then we either add the other Composables to the same file or other files depending on their complexity.
I don't say the approach here is wrong, I just think you should define a convention and stick to it 🙂.

Reply:
> In a compose-desktop app you won't have a fragment, or am I missing something?

That's correct 👍 .

> I don't say the approach here is wrong, I just think you should define a convention and stick to it 🙂.

Gotcha! Let's see what we're getting for a convention involving the feedback from @zwarm ,@RenanLukas & @mkevins 🙇.

My preference leans towards private composable functions defined in the same file but I'm not that much convinced it's really needed. Frankly I'm thinking the same with regards to defining a composable that is unlikely to be reused in a separate file, since that's making it "more public" by default but can't hurt in any way 😄 .